corehq/apps/locations/models.py

```diff
@@ -611,6 +619,23 @@ def sql_location(self):
         notify_of_deprecation("'sql_location' was just called on a sql_location.  That's kinda silly.")
         return self
 
+    @classmethod
+    def location_and_descendants(cls, location_ids):
+        locations = []
+        for location_id in location_ids:
+            location = cls.by_location_id(location_id)
+            locations += location.get_descendants(include_self=True)
+        return locations
+
+    @classmethod
+    def location_and_descendants_ids(cls, location_ids):
+        all_locations = cls.location_and_descendants(location_ids)
+        return [location.get_id for location in all_locations]
+
+    @classmethod
+    def users_at_locations_and_descendants(cls, location_ids):
+        location_ids = SQLLocation.location_and_descendants_ids(location_ids)
+        return UserES().location(location_ids).run().hits
```

Comment:
> This method does not belong here. Since this returns ES hits have moved it to UserES. Is that the right place for this?

I think a standalone function in `dbaccessors` (or `esaccessors` if you prefer), but I don't feel strongly about it.

> I would still retain the method location_and_descendants_ids and move this logic there. I think it would come handy in future. And should leave it under SQLLocation and not dbaccessors?

Maybe `LocationManager` or a dbaccessor?  The way I think about this is:

* `SQLLocation` is the place for methods which operate on an instance of `SQLLocation`, stuff like `boston.get_descendants()`.
* `SQLLocation.objects` is the place for things which don't rely on a specific location, but return one or more locations.  For couch models, we typically make these `@classmethod`s on the `Document` class, but Django models use the model manager, so that's the convention I use for this stuff.  This is especially true for things like `accessible_to_user`, which is likely to be used as part of a more complex query.
* `dbaccessors` or `utils` are good places for things which feel more like single function calls than components of a querying library.

I'm looking at `dbaccessors.py` and `util.py` now, and I don't think we have a good distinction between the two.  I do see a bunch of "get users" functions involving locations in the locations dbaccessors file.

I just noticed `get_locations_and_children` in `util.py`, and it's identical to `location_and_descendants`.  I do think the model manager would also be a good place for that though, since it returns a queryset (and then you can easily slap on a `.location_ids()` at the end if that's what you want).